plugin/restore/mysql/mysql.go

```diff
@@ -123,6 +132,165 @@ func getPITRDatabaseName(database string, timestamp int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncArchivedBinlogFiles syncs the binlogs between the instance and `saveDir`, but exclude latest binlog
+func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Instance, saveDir string) error {
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+	if err != nil {
+		return err
+	}
+
+	latestBinlogFileOnServer, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+
+	// compare file sizes and names to decide which files to download
+	downloadedIndex := make(map[int]struct{})
+	for index, serverFile := range binlogFilesOnServer {
+		// We don't download the latest binlog in SyncBinlogs()
+		if serverFile.Name == latestBinlogFileOnServer.Name {
+			continue
+		}
+		for _, localFile := range binlogFilesLocal {
+			localFileName := localFile.Name()
+			if localFileName == serverFile.Name {
+				// binlog file exists on local
+				if localFile.Size() != serverFile.Size {
+					// File size not match, delete and then download it
+					if err := os.Remove(filepath.Join(saveDir, localFileName)); err != nil {
+						return fmt.Errorf("cannot remove %s, error: %w", localFileName, err)
+					}
+				} else {
+					// file size match, record it in downloadedIndex
+					downloadedIndex[index] = struct{}{}
+				}
+			}
+		}
+	}
+
+	// download the binlog files not recorded in downloadedIndex
+	for i, serverFile := range binlogFilesOnServer {
+		if _, ok := downloadedIndex[i]; ok {
+			continue
+		}
+		if err := r.downloadBinlogFile(ctx, instance, saveDir, serverFile); err != nil {
+			return fmt.Errorf("cannot sync binlog %s, error: %w", serverFile.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// SyncLatestBinlog syncs the latest binlog between the instance and `saveDir`
+func (r *Restore) SyncLatestBinlog(ctx context.Context, instance *api.Instance, saveDir string) error {
+	latest, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+	return r.downloadBinlogFile(ctx, instance, saveDir, *latest)
+}
+
+// downloadBinlogFile syncs the binlog specified by `meta` between the instance and local.
+func (r *Restore) downloadBinlogFile(ctx context.Context, instance *api.Instance, saveDir string, meta mysql.BinlogFile) error {
+	tmpFilePrefix := fmt.Sprintf("tmp_%d_", time.Now().UnixNano())
+	// TODO(zp): support ssl?
+	cmd := exec.CommandContext(ctx, filepath.Join(mysqlbinlogPath, "bin", "mysqlbinlog"),
+		meta.Name,
+		fmt.Sprintf("--read-from-remote-server --host=%s --port=%s --user=%s --password=%s", instance.Host, instance.Port, instance.Username, instance.Password),
+		"--raw",
+		// Note, recheck here when upgrade embedding mysqlbinlog binary
```

Comment:
We may need a test case here to ensure that the `mysqlbinlog` can accept the parameters.

Reply:
I think when we test the correctness of SyncXXX(), the test will test it as well.